### PR TITLE
support docset unique in json schema validation

### PR DESCRIPTION
**Pull request by OsmondJiang**
feature: https://ceapex.visualstudio.com/Engineering/_boards/board/t/OPS%20Experience%20Team/Stories/?workitem=166621

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5622)

### CODE REVIEW

File: test/docfx.Test/lib/JsonSchemaTest.cs [hunk 1]
```diff
@@ -371,5 +371,38 @@ public void TestJsonSchemaValidation(string schema, string json, string expected
             var actual = string.Join('\n', errors.Select(err => err.ToString().Replace("\\r", "")).OrderBy(err => err).ToArray()).Replace('"', '\'');
```
Comment (yufeih) [inline test/docfx.Test/lib/JsonSchemaTest.cs]: consider adding a spec test in `validation.yml` to cover the integration between docfx and `JsonSchemaValidator`

File: src/docfx/lib/schema/JsonSchemaValidator.cs [hunk 2]
```diff
@@ -457,6 +468,37 @@ private void ValidateDeprecated(JsonSchema schema, string name, JToken token, Li
             }
         }
 
+        private void ValidateDocsetUnique(JsonSchema schema, string name, JObject map)
+        {
+            foreach (var docsetUniqueKey in schema.DocsetUnique)
+            {
+                if (map.TryGetValue(docsetUniqueKey, out var value))
+                {
+                    var key = string.IsNullOrEmpty(name) ? docsetUniqueKey : $"{name}.{docsetUniqueKey}";
```
Comment (yufeih) [inline src/docfx/lib/schema/JsonSchemaValidator.cs]: `(name, docsetUniqueKey)` may not be the correct property identifier, consider use `(schema, docsetUniqueKey)` as the key to identify a property. 

But still need `name` in line 497 to customize error message based on attribute name.

Comment (OsmondJiang) [reply src/docfx/lib/schema/JsonSchemaValidator.cs]: some properties defined in the DocsetUnique may not have schema.

File: src/docfx/build/metadata/MetadataProvider.cs [hunk 3]
```diff
@@ -87,6 +87,18 @@ internal class MetadataProvider
...
 
+        public List<Error> Validate()
+        {
+            var errors = new List<Error>();
+            foreach (var validator in _schemaValidators)
+            {
+                errors.AddRange(validator.PostValidate());
+            }
+
+            // todo: exclude files with errors from output
```
Comment (yufeih) [inline src/docfx/build/metadata/MetadataProvider.cs]: With #5621 merged, can remove this TODO:

Comment (OsmondJiang) [reply src/docfx/build/metadata/MetadataProvider.cs]: great!

**Pull request merged**